src/npfctl/npfctl.c

```diff
@@ -275,6 +278,96 @@ npfctl_print_addrmask(int alen, const char *fmt, const npf_addr_t *addr,
 	return buf;
 }
 
+static int
+npfctl_table_type(const char *typename)
+{
+	int i, type = 0;
+
+	static const struct tbltype_s {
+		const char *name;
+		u_int type;
+	} tbltypes[] = {
+		{ "ipset",	NPF_TABLE_IPSET	},
+		{ "lpm",	NPF_TABLE_LPM	},
+		{ "const",	NPF_TABLE_CONST	},
+		{ NULL,		0		}
+	};
+
+	for (i = 0; tbltypes[i].name != NULL; i++) {
+		if (strcmp(typename, tbltypes[i].name) != 0) {
+			continue;
+		}
+		type = tbltypes[i].type;
+		break;
+	}
+
+	return type;
+}
+
+__dead static void
+npfctl_table_replace(int fd, int argc, char **argv)
+{
+	const char *name, *newname, *fname;
+	nl_config_t *ncf;
+	nl_table_t *t;
+	npf_error_t errinfo;
+	int c, tid;
+	u_int type = 0;
+
+	name = newname = argv[0];
+	optind = 2;
+	while ((c = getopt(argc, argv, "n:t:")) != -1) {
```

Comment:
Not sure if these options will ever be useful in `npfctl`, but.. I guess why not.

Reply:
I thought they might be useful; for example if the user has renamed or changed type of a table in `/etc/npf.conf` and wants to keep their running config consistent (ie. for other admins who may try to add/remove addresses from tables).

I definitely think they need to be optional however; allowing a simple replace of a named table with the contents of a file. That's one of the reasons why I didn't use the config parser to handle parsing table details (as `npfctl_rule()` does).